File: app/src/main/java/com/esorokin/lantector/utils/ViewUtils.java
package com.esorokin.lantector.utils;

import android.app.Activity;
import android.content.Context;
import android.content.res.Resources;
import android.view.inputmethod.InputMethodManager;

public final class ViewUtils {
	private static final float DENSITY_FACTOR = 160f;

	private ViewUtils() {
		//utils
	}

	public static float pxToDp(float px) {
		float densityDpi = Resources.getSystem().getDisplayMetrics().densityDpi;
		return px / (densityDpi / DENSITY_FACTOR);
	}

	public static int dpToPx(int dp) {
		float density = Resources.getSystem().getDisplayMetrics().density;
		return Math.round(dp * density);
	}

	public static void hideKeyboard(Activity activity) {
		InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
		imm.hideSoftInputFromWindow(activity.getWindow().getDecorView().getWindowToken(), 0);
	}
}

File: app/src/main/java/com/esorokin/lantector/utils/ViewUtils.kt
package com.esorokin.lantector.utils

import android.app.Activity
import android.content.Context
import android.content.res.Resources
import android.view.inputmethod.InputMethodManager

object ViewUtils {
    private const val DENSITY_FACTOR = 160f

    fun pxToDp(px: Float): Float {
        val densityDpi = Resources.getSystem().displayMetrics.densityDpi.toFloat()
        return px / (densityDpi / DENSITY_FACTOR)
    }

    fun dpToPx(dp: Int): Int {
        val density = Resources.getSystem().displayMetrics.density
        return Math.round(dp * density)
    }

    fun hideKeyboard(activity: Activity) {
        val imm = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(activity.window.decorView.windowToken, 0)
    }
}
